cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -79,14 +80,29 @@ func NewCertificateAuthority(config *CertConfig) (*x509.Certificate, crypto.Sign
 	return cert, key, nil
 }
 
+// NewIntermediateCertificateAuthority creates new certificate and private key for an intermediate certificate authority
+func NewIntermediateCertificateAuthority(parentCert *x509.Certificate, parentKey crypto.Signer, config *CertConfig) (*x509.Certificate, crypto.Signer, error) {
```

Comment:
> Change made in this commit:

latest commit still has `config` as the last function argument.

should be:
```
NewIntermediateCertificateAuthority(config *CertConfig, parentCert *x509.Certificate, parentKey crypto.Signer)...
```

Reply:
Sorry, the comment about the change being made was intended for the `isCA` bool being moved to be the last argument. I've edited the comments so that it makes sense now.

As for moving `config` to the last argument, I've gone with this ordering for consistency with th existing function [`NewCertAndKey`](https://github.com/kubernetes/kubernetes/blob/b9ce4ac212d150212485fa29d62a2fbd783a57b0/cmd/kubeadm/app/util/pkiutil/pki_helpers.go#L83). Let me know if you would prefer it to be consistent, different or whether I should update both.

Reply:
ok, let's leave the ordering as is, for consistency with NewCertAndKey.
the pkiutils package has no API guaranties yet, but we may have to review some of these semantics at some point.